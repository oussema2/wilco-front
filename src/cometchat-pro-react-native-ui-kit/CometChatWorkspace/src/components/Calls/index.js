export { default as CometChatIncomingCall } from './CometChatIncomingCall';
export { default as CometChatOutgoingCall } from './CometChatOutgoingCall';
export { default as CometChatOutgoingDirectCall } from './CometChatOutgoingDirectCall';
export { default as CometChatIncomingDirectCall } from './CometChatIncomingDirectCall';
