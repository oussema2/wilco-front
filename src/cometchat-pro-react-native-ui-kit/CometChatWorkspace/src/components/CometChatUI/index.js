/* eslint-disable import/no-duplicates */
import React, { useEffect, useRef, useState } from 'react';
import { createBottomTabNavigator } from '@react-navigation/bottom-tabs';
import { View, Text } from 'react-native';
import { CometChatGroupListWithMessages } from '../Groups';
import { CometChatUserListWithMessages } from '../Users';
import { CometChatConversationListWithMessages } from '../Chats';
import { CometChatContextProvider } from '../../utils/CometChatContext';
import { CometChatUserProfile } from '../UserProfile';
import Ionicons from 'react-native-vector-icons/Ionicons';
import MaterialIcons from 'react-native-vector-icons/MaterialIcons';
import MCIIcons from 'react-native-vector-icons/MaterialCommunityIcons';
import theme from '../../resources/theme';
import { heightRatio } from '../../utils/consts';


const Tab = createBottomTabNavigator();

function CometChatUI() {
  const [tabs, setTabs] = useState(null);
  const contextRef = useRef(null);

  useEffect(() => {
    checkRestrictions();
  }, []);
  const checkRestrictions = async () => {
    let isChatEnabled = await contextRef.current.state.FeatureRestriction.isRecentChatListEnabled();
    let isGroupListEnabled = await contextRef.current.state.FeatureRestriction.isGroupListEnabled();
    let isUserSettingsEnabled = await contextRef.current.state.FeatureRestriction.isUserSettingsEnabled();
    let isUserListEnabled = await contextRef.current.state.FeatureRestriction.isUserListEnabled();
    let isCallListEnabled = await contextRef.current.state.FeatureRestriction.isCallListEnabled();
    setTabs({
      isChatEnabled,
      isGroupListEnabled,
      isUserSettingsEnabled,
      isUserListEnabled,
      isCallListEnabled,
    });
  };
  return (
    <CometChatContextProvider ref={contextRef}>
      {tabs ? (
        <Tab.Navigator
          screenOptions={({ route }) => ({
            tabBarIcon: ({ color }) => {
              let iconName;

              if (route.name === 'Chats') {
                return (
                  <MCIIcons name="chat" size={25 * heightRatio} color={color} />
                );
              }
              if (route.name === 'More') {
                return (
                  <MaterialIcons
                    name="more-horiz"
                    size={25 * heightRatio}
                    color={color}
                  />
                );
              }
              if (route.name === 'Users') {
                iconName = 'ios-person-circle-sharp';
              } else if (route.name === 'Groups') {
                iconName = 'people';
              }

              // You can return any component that you like here!
              return (
                <Ionicons
                  name={iconName}
                  size={24 * heightRatio}
                  color={color}
                />
              );
            },
          })}
          tabBarOptions={{
            activeTintColor: theme.color.blue,
            inactiveTintColor: 'rgba(0,0,0,0.5)',
            activeBackgroundColor: theme.color.white,
            inactiveBackgroundColor: theme.color.white,
            labelStyle: { fontSize: 12 },
          }}>
          {tabs.isChatEnabled && (
            <Tab.Screen
              name="Chats"
              component={CometChatConversationListWithMessages}
            />
          )}
          {tabs.isUserListEnabled && (
            <Tab.Screen
              name="Users"
              component={CometChatUserListWithMessages}
            />
          )}
          {tabs.isGroupListEnabled && (
            <Tab.Screen
              name="Groups"
              component={CometChatGroupListWithMessages}
            />
          )}
          {tabs.isUserSettingsEnabled && (
            <Tab.Screen name="More" component={CometChatUserProfile} />
          )}
        </Tab.Navigator>
      ) : null}
    </CometChatContextProvider>
  );
}

export default CometChatUI;
