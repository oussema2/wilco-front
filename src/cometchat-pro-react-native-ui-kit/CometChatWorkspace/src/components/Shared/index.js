export { default as CometChatAvatar } from './CometChatAvatar';
export { default as CometChatSharedMedia } from './CometChatSharedMedia';
export { default as CometChatBadgeCount } from './CometChatBadgeCount';
export { default as CometChatBackdrop } from './CometChatBackdrop';
export { default as CometChatUserPresence } from './CometChatUserPresence';
