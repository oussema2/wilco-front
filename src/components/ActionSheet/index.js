export { default as ActionSheet } from './ActionSheet';
