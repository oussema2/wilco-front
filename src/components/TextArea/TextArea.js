import React from 'react';
import PropTypes from 'prop-types';
import { InputWrapper } from '../InputWrapper';
import { MultilineInput } from '../MultilineInput';
import MentionInput from '../MentionInput/MentionInput';

const TextArea = ( {
	testID, textInputStyle, required, error, minimumLines, inputProps, capitalizeFirstLetter,
	supportMention, maxLength, popover, onFocus, bold
} ) => (
	<InputWrapper
		testID={testID}
		containerStyle={textInputStyle}
		required={required}
		error={error}
		Input={supportMention ? MentionInput : MultilineInput}
		minimumLines={minimumLines}
		capitalizeFirstLetter={capitalizeFirstLetter}
		maxLength={maxLength}
		popover={popover}
		bold={bold}
		{...inputProps}
		onFocus={onFocus || inputProps.onFocus}
	/>
);

TextArea.propTypes = {
	textInputStyle: PropTypes.instanceOf( Object ),
	testID: PropTypes.string,
	error: PropTypes.string,
	required: PropTypes.bool,
	minimumLines: PropTypes.number,
	inputProps: PropTypes.instanceOf( Object ),
	capitalizeFirstLetter: PropTypes.bool,
	supportMention: PropTypes.bool,
	maxLength: PropTypes.number,
	popover: PropTypes.node,
	onFocus: PropTypes.any,
	bold: PropTypes.bool
};

TextArea.defaultProps = {
	textInputStyle: {},
	testID: 'textArea',
	error: '',
	required: false,
	minimumLines: 1,
	inputProps: { label: '' },
	capitalizeFirstLetter: false,
	supportMention: false,
	maxLength: undefined,
	popover: null,
	onFocus: null,
	bold: true
};

export default TextArea;
