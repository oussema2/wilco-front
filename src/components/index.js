export { PrimaryButton } from './PrimaryButton';
export { TextInput } from './TextInput';
