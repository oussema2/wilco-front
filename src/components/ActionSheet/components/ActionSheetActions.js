import React from 'react';
import { observer } from 'mobx-react';
import { styles } from '../ActionSheet.styles';
import { ActionSheetButton } from '../../ActionSheetButton';
import { ActionSheetGradient } from '../../ActionSheetGradient';
import { useActionSheetService } from '../../../providers/ActionSheetProvider';

const ActionSheetActions = () => {
	const actionSheetService = useActionSheetService();

	return (
		<ActionSheetGradient style={styles.actions}>
			{actionSheetService.actions.map( ( action ) => (
				<ActionSheetButton
					key={action.title}
					{...action}
					onPress={() => {
						actionSheetService.close();
						action.onPress();
					}}
				/>
			) )}
		</ActionSheetGradient>
	);
};

export default observer( ActionSheetActions );
