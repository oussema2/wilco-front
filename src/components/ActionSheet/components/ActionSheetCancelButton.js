import React from 'react';
import { observer } from 'mobx-react';

import { View } from 'react-native';
import { styles } from '../ActionSheet.styles';
import { ActionSheetButton } from '../../ActionSheetButton';
import { useActionSheetService } from '../../../providers/ActionSheetProvider';

const ActionSheetCancelButton = () => {
	const actionSheetService = useActionSheetService();

	return (
		<View style={styles.cancelButtonContainer}>
			<ActionSheetButton
				title="Cancel"
				onPress={() => actionSheetService.close()}
			/>
		</View>
	);
};

export default observer( ActionSheetCancelButton );
