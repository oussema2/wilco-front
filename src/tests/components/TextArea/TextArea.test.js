import * as React from 'react';
import { fireEvent, render } from '@testing-library/react-native';

import { TextArea } from '../../../components/TextArea';

describe( 'TextArea', () => {
	describe( 'with the default props', () => {
		it( 'renders the TextArea component correctly', async () => {
			const component = render( <TextArea /> );
			expect( component ).toMatchSnapshot();
		} );
	} );

	describe( 'with a label', () => {
		it( 'renders the TextArea component correctly', async () => {
			const label = 'Label';
			const { getByTestId } = render( <TextArea inputProps={{ label }} /> );
			expect( getByTestId( 'input-label' ).props.children ).toBe( label );
			expect( getByTestId( 'input' ) ).not.toBeNull();
			expect( getByTestId( 'textArea' ) ).not.toBeNull();
		} );
	} );

	describe( 'when the value prop is set', () => {
		it( 'displays the value', () => {
			const value = 'Test value';
			const { getByTestId } = render( <TextArea inputProps={{ value }} /> );
			expect( getByTestId( 'input' ).props.value ).toBe( value );
		} );
	} );

	describe( 'when the disabled prop is set', () => {
		it( 'disables the input', () => {
			const { getByTestId } = render( <TextArea inputProps={{ disabled: true }} /> );
			expect( getByTestId( 'input' ).props.editable ).toBe( false );
		} );
	} );

	describe( 'when the required prop is set', () => {
		it( 'renders the TextArea component with a mandatory mark next to the label', () => {
			const component = render( <TextArea required /> );
			expect( component ).toMatchSnapshot();
			expect( component.getByTestId( 'input-required' ) ).not.toBeNull();
		} );
	} );

	describe( 'when the onFocus prop is set', () => {
		describe( 'and the text input is focused', () => {
			it( 'is calls the passed onFocus callback', () => {
				const onFocusMock = jest.fn();
				const { getByTestId } = render( <TextArea inputProps={{ onFocus: onFocusMock }} /> );
				fireEvent( getByTestId( 'input' ), 'focus' );
				expect( onFocusMock ).toHaveBeenCalledTimes( 1 );
			} );
		} );
	} );

	describe( 'when the onBlur prop is set', () => {
		describe( 'and the text input is blurred', () => {
			it( 'is calls the passed onBlur callback', () => {
				const onBlurMock = jest.fn();
				const { getByTestId } = render( <TextArea inputProps={{ onBlur: onBlurMock }} /> );
				fireEvent( getByTestId( 'input' ), 'blur' );
				expect( onBlurMock ).toHaveBeenCalledTimes( 1 );
			} );
		} );
	} );

	describe( 'when the onChange prop is set', () => {
		it( 'is passed to the TextArea element', () => {
			const onChange = () => {};
			const { getByTestId } = render( <TextArea inputProps={{ onChange }} /> );
			expect( getByTestId( 'input' ).props.onChangeText ).toBe( onChange );
		} );
	} );
} );
