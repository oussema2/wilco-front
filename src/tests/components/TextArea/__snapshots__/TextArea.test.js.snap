// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextArea when the required prop is set renders the TextArea component with a mandatory mark next to the label 1`] = `
<View
  style={
    Object {
      "minHeight": 98,
    }
  }
  testID="textArea"
>
  <View
    style={
      Object {
        "alignContent": "center",
        "alignItems": "flex-start",
        "display": "flex",
        "flexDirection": "row",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#242424",
          "fontFamily": "NotoSansJP-Black",
          "fontSize": 16,
          "fontWeight": undefined,
          "letterSpacing": -0.3,
          "lineHeight": 22,
          "marginBottom": 4,
        }
      }
      testID="input-label"
    >
      
    </Text>
    <Text
      style={
        Object {
          "color": "#D60A50",
          "fontFamily": "NotoSansJP-Bold",
          "fontSize": 14,
          "fontWeight": undefined,
          "letterSpacing": 0,
          "lineHeight": 20,
          "marginLeft": 4,
        }
      }
      testID="input-required"
    >
      *
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <TextInput
      autoCapitalize="none"
      editable={true}
      multiline={true}
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholder=""
      placeholderTextColor="#A0A0A0"
      scrollEnabled={false}
      selectionColor="#12A3BE"
      style={
        Object {
          "backgroundColor": "#F2F2F2",
          "borderRadius": 8,
          "color": "#000000",
          "fontFamily": "NotoSansJP-Bold",
          "fontSize": 14,
          "fontWeight": "normal",
          "letterSpacing": 0,
          "lineHeight": 20,
          "minHeight": 48,
          "paddingBottom": 16,
          "paddingHorizontal": 12,
          "paddingTop": 16,
          "textAlignVertical": "top",
        }
      }
      testID="input"
      value=""
    />
  </View>
</View>
`;

exports[`TextArea with the default props renders the TextArea component correctly 1`] = `
<View
  style={
    Object {
      "minHeight": 98,
    }
  }
  testID="textArea"
>
  <View
    style={
      Object {
        "alignContent": "center",
        "alignItems": "flex-start",
        "display": "flex",
        "flexDirection": "row",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#242424",
          "fontFamily": "NotoSansJP-Black",
          "fontSize": 16,
          "fontWeight": undefined,
          "letterSpacing": -0.3,
          "lineHeight": 22,
          "marginBottom": 4,
        }
      }
      testID="input-label"
    >
      
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <TextInput
      autoCapitalize="none"
      editable={true}
      multiline={true}
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholder=""
      placeholderTextColor="#A0A0A0"
      scrollEnabled={false}
      selectionColor="#12A3BE"
      style={
        Object {
          "backgroundColor": "#F2F2F2",
          "borderRadius": 8,
          "color": "#000000",
          "fontFamily": "NotoSansJP-Bold",
          "fontSize": 14,
          "fontWeight": "normal",
          "letterSpacing": 0,
          "lineHeight": 20,
          "minHeight": 48,
          "paddingBottom": 16,
          "paddingHorizontal": 12,
          "paddingTop": 16,
          "textAlignVertical": "top",
        }
      }
      testID="input"
      value=""
    />
  </View>
</View>
`;
